Secure mode

DarrylWong opened the pull request:
WIP: just trying to look at types of failures if we enable secure mode

DarrylWong committed:
turn on secure mode

WIP: just trying to see types of failures we get with secure mode

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/116278)
<!-- Reviewable:end -->

DarrylWong committed:
add passwordauth param to DefaultPGUrl

<what was there before: Previously, ...>
<why it needed to change: This was inadequate because ...>
<what you did about it: To address this, this patch ...>

DarrylWong committed:
add pgurlopts

<what was there before: Previously, ...>
<why it needed to change: This was inadequate because ...>
<what you did about it: To address this, this patch ...>

DarrylWong committed:
create roach certs

DarrylWong commented:
Closing and rebasing into smaller commits
